toolchain/check/import_ref.cpp

```diff
@@ -651,24 +669,28 @@ class ImportRefResolver {
               import_ir_.insts().TryGetAs<SemIR::AddrPattern>(bind_id)) {
         bind_id = addr->inner_id;
       }
-      GetLocalConstantId(bind_id);
+      auto bind_const_id =
+          import_ir_.insts().Is<SemIR::BindSymbolicName>(bind_id)
```

Comment:
I'll note, this looks slightly redundant with the above TryGetAs. If the original bind_id isn't an AddrPattern, it'll fetch the same inst twice. I think that's the common case, too.

You might consider something like:

```
auto bind_inst = import_ir_.insts().Get(bind_id);
if (auto addr = bind_inst.TryAs<SemIR::AddrPattern>()) {
  bind_id = addr->inner_id;
  bind_inst = import_ir_.insts().Get(bind_id);
}
auto bind_const_id = bind_inst.Is<SemIR::BindSymbolicName>()
```

Although maybe this will be in the noise...

Reply:
Done.

I was curious whether this would make any difference, so I took a look at the assembly. Looks like `insts().Get` is generally not getting inlined (probably because the `CHECK` makes it quite large), so my conclusion is... yes, this probably helps, but I'm sad that it's needed. Maybe we should use a `DCHECK` in `ValueStore::Get`?